src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/X500NameEncoder.ManagedDecode.cs

```diff
@@ -57,7 +57,7 @@ internal static partial class X500NameEncoder
                 // Throw in some "maybe-I-need-to-quote-this" quotes, and a couple of extra/extra-long O/OU values
                 // and round that up to the next programmer number, and you get that 512 should avoid reallocations
                 // in all but the most dire of cases.
-                StringBuilder decodedName = new StringBuilder(512);
+                ValueStringBuilder decodedName = new ValueStringBuilder(stackalloc char[512]);
```

Comment:
It's 1024 bytes of stack space, it's more than usual limits of 256 bytes?
Shouuld be OK, just to double-check.

Reply:
Yeah, it's on the steep side. I think 1024 bytes is the absolute max, and we've done that [elsewhere](https://github.com/dotnet/runtime/blob/f14aa5d03e8bbf7e7700e9cf021b91867fb7e775/src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpHeaders.cs#L243).